mitosheet/mitosheet/utils.py

```diff
@@ -82,7 +82,61 @@ def is_default_df_names(df_names: List[str]) -> bool:
     """
     return len(df_names) > 0 and df_names == [f'df{i + 1}' for i in range(len(df_names))]
 
+
+def get_conditonal_formatting_result(
+        state: StateType,
+        sheet_index: int,
+        df: pd.DataFrame,
+        conditional_formatting_rules: List[Dict[str, Any]],
+        max_rows: Optional[int]=MAX_ROWS,
+    ) -> ConditionalFormattingResult: 
+
+    df = df.head(max_rows)
+
+    invalid_conditional_formats: ConditionalFormattingInvalidResults = dict()
+    formatted_result: ConditionalFormattingCellResults = dict()
+
+    for conditional_format in conditional_formatting_rules:
+        format_uuid = conditional_format["format_uuid"]
+        column_ids = conditional_format["columnIDs"]
+
+        for column_id in column_ids:
+            # Skip columns this is not applied to
+            if column_id not in column_ids:
+                continue
+            
+            if column_id not in formatted_result:
+                formatted_result[column_id] = dict()
+
+            filters  = conditional_format["filters"]
+            backgroundColor = conditional_format.get("backgroundColor", None)
+            color = conditional_format.get("color", None)
+
+            column_header = state.column_ids.get_column_header_by_id(sheet_index, column_id)
+
+            # Use the get_applied_filter function from our filtering infrastructure
+            from mitosheet.step_performers.filter import get_full_applied_filter
+            try:
+                full_applied_filter, _ = get_full_applied_filter(df, column_header, 'And', filters)
+                applied_indexes = full_applied_filter[full_applied_filter].index.tolist()
+
+                for index in applied_indexes:
+                    # We need to make this index valid json, and do so in a way that is consistent with how indexes
+                    # are sent to the frontend
+                    json_index = json.dumps(index, cls=NpEncoder)
+                    formatted_result[column_id][json_index] = {'backgroundColor': backgroundColor, 'color': color}
```

Comment:
I think instead of doing these by dataframe indexes, it would be better to do it as mito row indexes. We already talked about this in the spec, so up to you. But note that there is an existing bug with doing it this way.

Reply:
I'm gonna leave this as is for now -- I think it's fine and reasonable to reference rows in this way. We can adjust this approach if it doesn't work going forward, but I think it's ok!